server/src/main/java/org/elasticsearch/index/shard/IndexReaderWrapper.java

```diff
@@ -87,22 +78,18 @@ protected IndexSearcher wrap(IndexSearcher searcher) throws IOException {
             }
         }
 
-        final IndexSearcher origIndexSearcher = engineSearcher.searcher();
-        final IndexSearcher innerIndexSearcher = new IndexSearcher(reader);
-        innerIndexSearcher.setQueryCache(origIndexSearcher.getQueryCache());
-        innerIndexSearcher.setQueryCachingPolicy(origIndexSearcher.getQueryCachingPolicy());
-        innerIndexSearcher.setSimilarity(origIndexSearcher.getSimilarity());
-        // TODO: Right now IndexSearcher isn't wrapper friendly, when it becomes wrapper friendly we should revise this extension point
-        // For example if IndexSearcher#rewrite() is overwritten than also IndexSearcher#createNormalizedWeight needs to be overwritten
-        // This needs to be fixed before we can allow the IndexSearcher from Engine to be wrapped multiple times
-        final IndexSearcher indexSearcher = wrap(innerIndexSearcher);
-        if (reader == nonClosingReaderWrapper && indexSearcher == innerIndexSearcher) {
+        if (reader == nonClosingReaderWrapper) {
```

Comment:
should we move this into indexShard instead of in some helper function?

Reply:
Agreed, I pushed https://github.com/elastic/elasticsearch/pull/43645/commits/615c48b31403c587880f5beaff871e7eae73901f